src/Og.php

```diff
@@ -192,6 +199,53 @@ public static function isGroupAudienceField(FieldDefinitionInterface $field_conf
   }
 
   /**
+   * Return all the group audience fields in an entity.
+   *
+   * @param string $entity_type_id
+   *   The entity type.
+   * @param string  $bundle
+   *   The bundle name to be checked.
+   * @param string $group_type_id
+   *   Filter list to only include fields referencing a specific group type.
+   * @param string $group_bundle
+   *   Filter list to only include fields referencing a specific group bundle.
+   *   Fields that do not specify any bundle restrictions at all are also
+   *   included.
+   *
+   * @return \Drupal\Core\Field\FieldDefinitionInterface[]
+   *   An array of field definitions, keyed by field name; Or an empty array if
+   *   none found.
+   */
+  public static function getAllGroupAudienceFields($entity_type_id = 'user', $bundle = 'user', $group_type_id = NULL, $group_bundle = NULL) {
+    $return = [];
```

Comment:
Yes. Can definitely be a follow up. What I am saying is that we will want
both static cache and real cache in a cache backend. This can potentially
get called many times during a request. That kind of fetching and
processing is not that cheap. If we cached and statically cached a simple
map. Similar to our group storage data.
On 29 Nov 2015 17:42, "Amitai Burstein" <notifications@github.com> wrote:

> In src/Og.php
> <https://github.com/amitaibu/og/pull/46#discussion_r46095375>:
>
> > +   *   The entity type.
> > +   * @param string  $bundle
> > +   *   The bundle name to be checked.
> > +   * @param string $group_type_id
> > +   *   Filter list to only include fields referencing a specific group type.
> > +   * @param string $group_bundle
> > +   *   Filter list to only include fields referencing a specific group bundle.
> > +   *   Fields that do not specify any bundle restrictions at all are also
> > +   *   included.
> > +   *
> > +   * @return \Drupal\Core\Field\FieldDefinitionInterface[]
> > +   *   An array of field definitions, keyed by field name; Or an empty array if
> > +   *   none found.
> > +   */
> > +  public static function getAllGroupAudienceFields($entity_type_id = 'user', $bundle = 'user', $group_type_id = NULL, $group_bundle = NULL) {
> > +    $return = [];
>
> @damiankloip <https://github.com/damiankloip> not sure I understand your
> cache comment, but I think we both agree it should be a follow up, right?
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/amitaibu/og/pull/46/files#r46095375>.
>